apps/dashboard/src/main/java/com/akto/action/CustomDataTypeAction.java

```diff
@@ -236,10 +257,82 @@ public String saveAktoDataType(){
         }
 
         SingleTypeInfo.fetchCustomDataTypes(Context.accountId.get());
+        if(redacted){
+            int accountId = Context.accountId.get();
+            Runnable r = () ->{
+                Context.accountId.set(accountId);
+                loggerMaker.infoAndAddToDb("Triggered a job to fix existing data for " + aktoDataType.getName(), LogDb.DASHBOARD);
+                handleDataTypeRedaction();
+            };
```

Comment:
Please make a "class Task" for all such thread-operations. And remove code duplication

Reply:
I just need a function to be executed in a thread, 1 thing I have fixed here is used a thread pool instead of a simple thread.